libmamba/include/mamba/util/heap_optional.hpp

```diff
@@ -251,17 +251,4 @@ namespace mamba::util
     }
 }
 
-template <typename T>
-struct std::hash<mamba::util::heap_optional<T>>
-{
-    std::size_t operator()(const mamba::util::heap_optional<T>& opt) const
-    {
-        if (opt.has_value())
-        {
-            return std::hash<T>{}(*opt);
-        }
-        return 0;
-    }
-};
-
```

Comment:
This is not used anymore?

Reply:
Yes, this is not needed thanks to [this line](https://github.com/mamba-org/mamba/pull/3369/files#diff-7b3f6d24884e8d3e7e3a160ab570bdfe231bf8a13df239de5818e98581840ab5R1049) which hashes the value contained by the `heap_optional`.

IIRC, @Klaim has it in mind and I understood why.